spacy_curated_transformers/cli/fill_config_transformer.py

```diff
@@ -0,0 +1,438 @@
+import json
+import sys
+from enum import Enum
+from pathlib import Path
+from typing import Any, Dict, Optional, Tuple
+
+from confection import Config
+from spacy import util
+from spacy.cli._util import import_code, init_cli, parse_config_overrides
+from spacy.cli.init_config import save_config
+from typer import Argument as Arg
+from typer import Context
+from typer import Option as Opt
+from wasabi import Printer
+
+from .._compat import has_hf_transformers, has_huggingface_hub
+
+
+@init_cli.command(
+    "fill-config-transformer",
+    context_settings={"allow_extra_args": True, "ignore_unknown_options": False},
+)
+def init_fill_config_transformer_cli(
+    # fmt: off
+    ctx: Context,  # This is only used to read additional arguments
+    base_path: Path = Arg(..., help="Path to the base config file to fill", exists=True, allow_dash=True, dir_okay=False),
+    output_path: Path = Arg("-", help="Path to output .cfg file or '-' for stdout (default: stdout)", exists=False, allow_dash=True),
+    model_name: Optional[str] = Opt(None, "--model-name", "-m", help="Name of the Hugging Face model. If not provided, the model name will be read in from the encoder loader config"),
+    model_revision: Optional[str] = Opt(None, "--model-revision", "-r", help="Revision of the Huggin Face model. Must be provided with the model name."),
```

Comment:
I understand the motivation, but I think `"main"` would be a more helpful default. A warning about using `"main"` instead from our side would be fine?